src/providers/test-result/__tests__/test-result.spec.ts

```diff
@@ -61,137 +40,216 @@ describe('TestResultCalculatorProvider', () => {
+          done();
+        });
+      });
+      it('should return a Fail when there are 16 driving faults and a serious fault', (done) => {
+        const testData: CatBEUniqueTypes.TestData = {
+          seriousFaults: {
+            positioningNormalDriving: true,
+          },
+          drivingFaults: {
+            ...drivingFaults,
+            judgementMeeting: 1,
+          },
+        };
+        testResultProvider.calculateTestResult(TestCategory.BE, testData).subscribe((result) => {
+          expect(result).toBe(ActivityCodes.FAIL);
+          done();
+        });
       });
     });
     it('should return a Fail when there are 15 driving faults and a dangerous fault', (done) => {
-      // ARRANGE
-      const testData: CatBUniqueTypes.TestData = {
-        ...initialTestData,
+      const testData: CatBEUniqueTypes.TestData = {
         drivingFaults,
         dangerousFaults: {
           positioningNormalDriving: true,
         },
       };
-      // ASSERT
-      testResultProvider.calculateCatBTestResult(testData).subscribe((result) => {
+      testResultProvider.calculateTestResult(TestCategory.BE, testData).subscribe((result) => {
         expect(result).toBe(ActivityCodes.FAIL);
         done();
       });
     });
     it('should return a Fail when there are 15 driving faults and a serious fault', (done) => {
-      // ARRANGE
-      const testData: CatBUniqueTypes.TestData = {
-        ...initialTestData,
+      const testData: CatBEUniqueTypes.TestData = {
         drivingFaults,
         seriousFaults: {
           positioningNormalDriving: true,
         },
       };
-      // ASSERT
-      testResultProvider.calculateCatBTestResult(testData).subscribe((result) => {
+      testResultProvider.calculateTestResult(TestCategory.BE, testData).subscribe((result) => {
         expect(result).toBe(ActivityCodes.FAIL);
         done();
       });
     });
-
     it('should return a Pass when there are 15 driving faults', (done) => {
-      // ARRANGE
-      const testData: CatBUniqueTypes.TestData = {
-        ...initialTestData,
+      const testData: CatBEUniqueTypes.TestData = {
         drivingFaults,
       };
-      // ASSERT
-      testResultProvider.calculateCatBTestResult(testData).subscribe((result) => {
+      testResultProvider.calculateTestResult(TestCategory.BE, testData).subscribe((result) => {
         expect(result).toBe(ActivityCodes.PASS);
         done();
       });
     });
     it('should return a Pass when there are no driving faults', (done) => {
-      // ARRANGE
-      const testData: CatBUniqueTypes.TestData = {
-        ...initialTestData,
-      };
-      // ASSERT
-      testResultProvider.calculateCatBTestResult(testData).subscribe((result) => {
+      const testData: CatBEUniqueTypes.TestData = {};
+      testResultProvider.calculateTestResult(TestCategory.BE, testData).subscribe((result) => {
         expect(result).toBe(ActivityCodes.PASS);
         done();
       });
```

Comment:
think there should be a test for anything over 15 driving faults with no serious and no dangerous.

Reply:
There is a test for that `should return a Fail when there are 16 driving faults'`